### PR TITLE
[LOOP-3798] Manual glucose entry uses display glucose units

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-3798

### CODE REVIEW

File: Loop/Views/BolusEntryView.swift [hunk 1]
```diff
@@ -74,7 +73,7 @@ struct BolusEntryView: View {
                 // The view model can disable manual glucose entry if CGM data returns.
                 if !isManualGlucoseEntryEnabled {
                     self.isManualGlucoseEntryRowVisible = false
-                    self.enteredManualGlucose = ""
+                    self.viewModel.manualGlucoseEntry = ""
```
Comment (rickpasetto) [inline Loop/Views/BolusEntryView.swift]: Now that `manualGlucoseEntry` is in the `ViewModel`, it seems more appropriate to me that the policy of clearing it belongs there and not here

File: Loop/View Models/BolusEntryViewModel.swift [hunk 2]
```diff
@@ -116,6 +118,52 @@ final class BolusEntryViewModel: ObservableObject {
...
+            // only update enteredManualGlucose if needed
+            if enteredManualGlucose == nil ||
+                glucoseAmount != glucoseQuantityFormatter.string(from: enteredManualGlucose!, for: cachedDisplayGlucoseUnit, includeUnit: false)
+            {
+                self.enteredManualGlucose = HKQuantity(unit: cachedDisplayGlucoseUnit, doubleValue: newGlucoseValue)
+            }
+        }
+    }
+
+    var manualGlucoseEntry: String {
```
Comment (rickpasetto) [inline Loop/View Models/BolusEntryViewModel.swift]: [nit] I'm confusing this with `enteredManualGlucose` and `glucoseAmount`.  Maybe consider adding some code comments to explain what these two are and how they are different?

File: Loop/View Models/SimpleBolusViewModel.swift [hunk 3]
```diff
@@ -76,10 +76,34 @@ class SimpleBolusViewModel: ObservableObject {
...
+            }
+
+            return glucoseAmount
+        }
+        set {
+            glucoseAmount = newValue
+        }
+    }
+
+    @Published var glucoseAmount: String = "" {
```
Comment (rickpasetto) [inline Loop/View Models/SimpleBolusViewModel.swift]: [nit] Similar here, consider adding code comments on what the difference between `enteredGlucoseAmount` and `glucoseAmount` is.

File: Loop/View Models/BolusEntryViewModel.swift [hunk 4]
```diff
@@ -116,6 +118,52 @@ final class BolusEntryViewModel: ObservableObject {
...
+        get {
```
Comment (rickpasetto) [inline Loop/View Models/BolusEntryViewModel.swift:148]: unit tests for new code?

**Review by rickpasetto — APPROVED** [review]
Couple of non-blocking suggestions, other than that LGTM

File: Loop/View Models/BolusEntryViewModel.swift (hunk 2 above)
Comment (nhamming) [reply Loop/View Models/BolusEntryViewModel.swift]: Yes, even I find this confusing :) Just occurred to me that changing `glucoseAmount` -> `_manualGlucoseEntry` is closer to the intention, which is `_manualGlucoseEntry` backs `manualGlucoseEntry`.

**Pull request merged**